specs/drafts/genesis-v2.md

```diff
@@ -0,0 +1,174 @@
+  ## The FQN (fully qualified name) of the parent subnet in the IPC hierarchy, using standardised slash notation, e.g. /r314/f410.../f410...
+  ## Or null if this subnet is standalone (unrooted).
+  parent: string | null
+  ## The FQN (fully qualified name) of this subnet in the IPC hierarchy, using standardised slash notation, e.g. /r314/f410.../f410...
+  ## If this subnet is standalone (unrooted), this is a custom string.
+  path: string
+  ## A user-defined alias for this subnet.
+  alias: string
+  ## Genesis timestamp, as an ISO8601 string.
+  genesis_timestamp: string
+  ## Genesis network version.
+  network_version: uint64 
+  
+## Consensus configuration for this subnet.
+consensus:
+  ## The consensus engine adopted by this subnet. The only value supported is 'tendermint', denoting that CometBFT will
+  ## be used.
+  engine: 'tendermint'
+  ## Validation configuration.
+  validation:
+    ## The percentage of power/weight above which 
+    majority_percentage: uint64
+    ## Validator membership configuration.
+    membership:
+      ## Defines how membership is determined.
+      ## - 'static': members and weights are defined upfront at subnet creation time, and cannot be changed after genesis.
+      ## - 'collateral': members and weights are projected from collateral deposited at the parent (proof of stake style).
+      ## - 'federated': members and weights are determined arbitrarily by an authority at the parent (proof of authority style).
+      mode: static | collateral | federated
+      ## The scaling factor that applies to the raw power communicated to the parent (raw * 1e-{power_scale}). 
+      ## To be deprecated during contracts refactor project.
+      power_scale: uint8
+      ## The maximum number of active validators at a time.
+      ## To be deprecated during contracts refactor project.
+      max_active: uint64
+    ## Initial validator set as an array of pubkeys and weights.
+    initial_members:
+      - pubkey: bytes
+        weight: uint64
+  
+## Balances to be credited to the specified addresses at genesis, directly on the state tree,
+## the precision unit of the circulating supply source.
+balances:
+  [address]: uint256
+ 
+## Execution layer configuration. 
+execution:
+  ## Gas market configuration
+  gas_market:
+    ## Oscillation function. Values: eip1559 or static.
+    kind: eip1559 | static
+    ## Initial base fee.
+    initial_base_fee: uint256 
+  ## Configuration of FVM runtimes.
+  runtimes:
+    ## Filecoin EVM configuration.
+    fevm:
+      ## Contract deployment permissions.
+      contract_deployment_permissions:
+        ## Contract deployment permissioning mode. Values: 'unrestricted', or 'allowlist'.
+        mode: unrestricted | allowlist
+        ## If the mode is 'allowlist', specify a list of allowed addresses, or blank if none is allowed.
+        allowlist:
+          - addresses
+         
+## CIDs of actor manifests for built-in actors and for custom actors.
+actor_bytecode:
```

Comment:
so each time after processing, the data will be written to `blockstore`?

Reply:
So when you update the genesis spec to add the bytecodes, you would be operating on the blockstore, yeah. The key idea here is that you import the genesis blockstore into the node as-is.